core/src/main/java/org/elasticsearch/index/engine/SegmentsStats.java

```diff
@@ -292,6 +306,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.byteSizeField(Fields.INDEX_WRITER_MEMORY_IN_BYTES, Fields.INDEX_WRITER_MEMORY, indexWriterMemoryInBytes);
         builder.byteSizeField(Fields.VERSION_MAP_MEMORY_IN_BYTES, Fields.VERSION_MAP_MEMORY, versionMapMemoryInBytes);
         builder.byteSizeField(Fields.FIXED_BIT_SET_MEMORY_IN_BYTES, Fields.FIXED_BIT_SET, bitsetMemoryInBytes);
+        builder.field(Fields.MAX_UNSAFE_AUTO_ID_TIMESTAMP, maxUnsafeAutoIdTimestamp);
```

Comment:
I think we should inline with what it is today. It's indeed an implementation detail of the engine so I will be 100% ok with changing the stats if we change the implementation on the engine level.